compiler/ETA/Main/DriverPipeline.hs

```diff
@@ -1283,25 +1298,38 @@ linkingNeeded dflags linkables pkgDeps = do
         -- modification times on all of the objects and libraries, then omit
         -- linking (unless the -fforce-recomp flag was given).
   let jarFile = jarFileName dflags
-  eJarTime <- tryIO $ getModificationUTCTime jarFile
+      getTime = tryIO . getModificationUTCTime
+  eJarTime <- getTime jarFile
   case eJarTime of
     Left _  -> return True
     Right t -> do
+        debugTraceMsg dflags 3 (text $ "linkingNeeded: previousJar time = " ++ show t)
         -- TODO: Factor in ldOptions too
         let jarTimes = map linkableTime linkables
-        if any (t <) jarTimes then return True
+        debugTraceMsg dflags 3 (text $ "linkingNeeded: linkablesJar times = " ++
+                                show jarTimes)
+        if any (t <) jarTimes then
+          return True
         else do
           let pkgHSLibs  = [ (libraryDirs c, lib)
                           | Just c <- map (lookupInstalledPackage dflags) pkgDeps
                           , lib <- packageHsLibs dflags c ]
           pkgLibFiles <- mapM (uncurry $ findHSLib dflags) pkgHSLibs
-          if any isNothing pkgLibFiles then
-            return True
+          if any isNothing pkgLibFiles then do
+             debugTraceMsg dflags 3 (text $ "linkingNeeded: any isNothing pkgLibFiles")
+             return True
           else do
-            eLibTimes <- mapM (tryIO . getModificationUTCTime)
-                              (catMaybes pkgLibFiles)
+            extras <- getExtrasFileName dflags
+            let -- Currently __extras.jar is always generated even when modules are not compiled
```

Comment:
Is this something that should be changed in the future? If so please file an issue so we can track it - thanks!

Reply:
Mmm the change could be hash the java and class files instead the final `__extras.jar` and avoid its own creation if the original files dont change. But now the eta packages usually have few java sources files and i think we would not reduce the build time a lot

Reply:
Agreed. Thanks for the explanation!